File: Bindings~/java/src/main/java/ro/altom/alttester/Commands/InputActions/AltMultiPointSwipe.java
package ro.altom.alttester.Commands.InputActions;

import ro.altom.alttester.IMessageHandler;
import ro.altom.alttester.Commands.AltBaseCommand;

/**
 * Similar command like swipe but instead of swipe from point A to point B you
 * are able to give list a points.
 */
public class AltMultiPointSwipe extends AltBaseCommand {

    private AltMultiPointSwipeParams params;

    public AltMultiPointSwipe(IMessageHandler messageHandler, AltMultiPointSwipeParams params) {
        super(messageHandler);
        this.params = params;
    }

    public void Execute() {
        SendCommand(params);
        String data = recvall(params, String.class);
        validateResponse("Ok", data);

        if (params.getWait()) {
            data = recvall(params, String.class);
            validateResponse("Finished", data);
        }
    }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltMultipointSwipe.cs
/*
    Copyright(C) 2024 Altom Consulting

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

namespace AltTester.AltTesterUnitySDK.Driver.Commands
{
    public class AltMultipointSwipe : AltBaseCommand
    {
        AltMultipointSwipeParams cmdParams;

        public AltMultipointSwipe(IDriverCommunication commHandler, AltVector2[] positions, float duration, bool wait) : base(commHandler)
        {
            cmdParams = new AltMultipointSwipeParams(positions, duration, wait);
        }

        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);

            if (cmdParams.wait)
            {
                data = CommHandler.Recvall<string>(cmdParams);
                ValidateResponse("Finished", data);
            }
        }
    }
}
